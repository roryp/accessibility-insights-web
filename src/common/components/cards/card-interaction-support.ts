// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export type CardInteractionSupport = {
    supportsHighlighting: boolean;
    supportsIssueFiling: boolean;
    supportsCopyFailureDetails: boolean;
};

export const allCardInteractionsSupported: CardInteractionSupport = {
    supportsHighlighting: true,
    supportsIssueFiling: true,
    supportsCopyFailureDetails: true,
};

export const noCardInteractionsSupported: CardInteractionSupport = {
    supportsHighlighting: false,
    supportsIssueFiling: false,
    supportsCopyFailureDetails: false,
};

export const onlyHighlightingSupported: CardInteractionSupport = {
    ...noCardInteractionsSupported,
    supportsHighlighting: true,
};

export const onlyUserConfigAgnosticCardInteractionsSupported: CardInteractionSupport = {
    ...allCardInteractionsSupported,
    supportsIssueFiling: false,
};
