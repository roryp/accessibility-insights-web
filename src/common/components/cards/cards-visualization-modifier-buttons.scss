// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
.cards-visualization-modifiers-container {
    display: flex;
    flex-wrap: wrap;
}
