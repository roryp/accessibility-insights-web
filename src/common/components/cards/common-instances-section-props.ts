// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ScanMetadata } from 'common/types/store-data/unified-data-interface';
import { CardsViewModel } from '../../types/store-data/card-view-model';
import { UserConfigurationStoreData } from '../../types/store-data/user-configuration-store';
import { ResultSectionDeps } from './result-section';

export type CommonInstancesSectionDeps = ResultSectionDeps;
export type CommonInstancesSectionProps = {
    deps: CommonInstancesSectionDeps;
    cardsViewData: CardsViewModel;
    userConfigurationStoreData: UserConfigurationStoreData;
    scanMetadata: ScanMetadata;
    shouldAlertFailuresCount?: boolean;
};
