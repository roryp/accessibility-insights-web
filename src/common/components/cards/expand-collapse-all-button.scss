// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
.expand-collapse-all-button {
    display: flex;
    padding: unset;
    padding-right: 27px;
    margin-left: -2px;
    margin-top: 2px;
}
