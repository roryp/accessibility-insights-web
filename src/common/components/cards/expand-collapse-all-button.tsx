// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { NamedFC } from 'common/react/named-fc';
import { ActionButton } from 'office-ui-fabric-react';
import * as React from 'react';
import * as styles from './expand-collapse-all-button.scss';

export type ExpandCollapseAllButtonDeps = {
    cardSelectionMessageCreator: CardSelectionMessageCreator;
};

export type ExpandCollapseAllButtonProps = {
    deps: ExpandCollapseAllButtonDeps;
    allCardsCollapsed: boolean;
};

export const ExpandCollapseAllButton = NamedFC<ExpandCollapseAllButtonProps>(
    'ExpandCollapseAllButton',
    props => {
        const { deps, allCardsCollapsed } = props;

        let expandCollapseAllButtonHandler = deps.cardSelectionMessageCreator.collapseAllRules;
        let buttonText = 'Collapse all';
        let iconName = 'ChevronDown';

        if (allCardsCollapsed) {
            expandCollapseAllButtonHandler = deps.cardSelectionMessageCreator.expandAllRules;
            buttonText = 'Expand all';
            iconName = 'ChevronRight';
        }

        return (
            <ActionButton
                iconProps={{ iconName }}
                onClick={expandCollapseAllButtonHandler}
                aria-expanded={!allCardsCollapsed}
                className={styles.expandCollapseAllButton}
            >
                {buttonText}
            </ActionButton>
        );
    },
);
