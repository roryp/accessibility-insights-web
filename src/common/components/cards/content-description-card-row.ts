// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GetLabelledStringPropertyCardRow } from 'common/components/cards/get-labelled-string-property-card-row';
export const ContentDescriptionCardRow = GetLabelledStringPropertyCardRow('Content description');
