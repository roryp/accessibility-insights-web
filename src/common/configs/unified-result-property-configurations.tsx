// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ClassNameCardRow } from 'common/components/cards/class-name-card-row';
import { ContentDescriptionCardRow } from 'common/components/cards/content-description-card-row';

import { TextCardRow } from 'common/components/cards/text-card-row';
import { HowToFixAndroidCardRow } from '../components/cards/how-to-fix-android-card-row';
import { HowToFixWebCardRow } from '../components/cards/how-to-fix-card-row';
import { PathCardRow } from '../components/cards/path-card-row';
import { SnippetCardRow } from '../components/cards/snippet-card-row';
import { FixInstructionProcessor } from '../components/fix-instruction-processor';
import { ReactFCWithDisplayName } from '../react/named-fc';

export type PropertyType =
    | 'css-selector'
    | 'how-to-fix-web'
    | 'snippet'
    | 'className'
    | 'contentDescription'
    | 'text'
    | 'howToFixFormat';
export const AllPropertyTypes: PropertyType[] = [
    'css-selector',
    'how-to-fix-web',
    'snippet',
    'className',
    'contentDescription',
    'text',
    'howToFixFormat',
];

export interface CardRowDeps {
    fixInstructionProcessor: FixInstructionProcessor;
}

export interface CardRowProps {
    deps: CardRowDeps;
    index: number;
    propertyData: any;
}

export interface PropertyConfiguration {
    cardRow: ReactFCWithDisplayName<CardRowProps>;
}

export const howToFixConfiguration: PropertyConfiguration = {
    cardRow: HowToFixWebCardRow,
};

export const howToFixAndroidConfiguration: PropertyConfiguration = {
    cardRow: HowToFixAndroidCardRow,
};

export const cssSelectorConfiguration: PropertyConfiguration = {
    cardRow: PathCardRow,
};

export const snippetConfiguration: PropertyConfiguration = {
    cardRow: SnippetCardRow,
};

export const classNameConfiguration: PropertyConfiguration = {
    cardRow: ClassNameCardRow,
};

export const contentDescriptionConfiguration: PropertyConfiguration = {
    cardRow: ContentDescriptionCardRow,
};

export const textConfiguration: PropertyConfiguration = {
    cardRow: TextCardRow,
};

export type PropertyIdToConfigurationMap = {
    [key in PropertyType]: PropertyConfiguration;
};
const propertyIdToConfigurationMap: PropertyIdToConfigurationMap = {
    'css-selector': cssSelectorConfiguration,
    'how-to-fix-web': howToFixConfiguration,
    howToFixFormat: howToFixAndroidConfiguration,
    snippet: snippetConfiguration,
    className: classNameConfiguration,
    contentDescription: contentDescriptionConfiguration,
    text: textConfiguration,
};

export function getPropertyConfiguration(id: string): Readonly<PropertyConfiguration> {
    return propertyIdToConfigurationMap[id];
}
