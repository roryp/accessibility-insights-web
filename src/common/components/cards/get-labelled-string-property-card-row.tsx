// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { isEmpty } from 'lodash';
import * as React from 'react';
import { CardRowProps } from '../../../common/configs/unified-result-property-configurations';
import { NamedFC } from '../../../common/react/named-fc';
import { SimpleCardRow } from './simple-card-row';

export interface StringPropertyCardRowProps extends CardRowProps {
    propertyData: string;
}

export const GetLabelledStringPropertyCardRow = (label: string, contentClassName?: string) => {
    return NamedFC<StringPropertyCardRowProps>('StringPropertyCardRowProps', props => {
        if (isEmpty(props.propertyData)) {
            return null;
        }

        return (
            <SimpleCardRow
                label={label}
                content={props.propertyData}
                rowKey={`${label}-${props.index}`}
                contentClassName={contentClassName}
            />
        );
    });
};
