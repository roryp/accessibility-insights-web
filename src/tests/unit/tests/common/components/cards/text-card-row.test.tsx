// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { StringPropertyCardRowProps } from 'common/components/cards/get-labelled-string-property-card-row';
import { TextCardRow } from 'common/components/cards/text-card-row';
import { CardRowDeps } from 'common/configs/unified-result-property-configurations';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('TextCardRow', () => {
    it('renders', () => {
        const props: StringPropertyCardRowProps = {
            propertyData: 'test text',
            deps: {} as CardRowDeps,
            index: -1,
        };

        const wrapped = shallow(<TextCardRow {...props} />);
        expect(wrapped.getElement()).toMatchSnapshot();
    });
});
