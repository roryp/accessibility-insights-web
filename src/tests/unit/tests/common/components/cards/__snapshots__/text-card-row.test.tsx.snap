// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextCardRow renders 1`] = `
<SimpleCardRow
  content="test text"
  label="Text"
  rowKey="Text--1"
/>
`;
